boards/st/stm32n6570_dk/doc/index.rst

```diff
@@ -0,0 +1,176 @@
+| GPIO      | on-chip    | gpio                                |
++-----------+------------+-------------------------------------+
+| NVIC      | on-chip    | nested vector interrupt controller  |
++-----------+------------+-------------------------------------+
+| UART      | on-chip    | serial port-polling;                |
+|           |            | serial port-interrupt               |
++-----------+------------+-------------------------------------+
+
+
+Other hardware features are not yet supported on this Zephyr port.
+
+The default configuration can be found in the defconfig file:
+:zephyr_file:`boards/st/stm32n6570_dk/stm32n6570_dk_defconfig`
+
+
+Connections and IOs
+===================
+
+STM32N6570_DK Board has 12 GPIO controllers. These controllers are responsible
+for pin muxing, input/output, pull-up, etc.
+
+For more details please refer to `STM32N6570_DK User Manual`_.
+
+Default Zephyr Peripheral Mapping:
+----------------------------------
+
+- LD1 : PO1
+- LD2 : PG10
+- USART_1_TX : PE5
+- USART_1_RX : PE6
+
+System Clock
+------------
+
+STM32N6570_DK System Clock could be driven by internal or external oscillator,
+as well as main PLL clock. By default System clock is driven by PLL clock at
+400MHz, driven by 64MHz high speed internal oscillator.
+
+Serial Port
+-----------
+
+STM32N6570_DK board has 10 U(S)ARTs. The Zephyr console output is assigned to
+USART1. Default settings are 115200 8N1.
+
+Programming and Debugging
+*************************
+
+STM32N6570_DK board includes an ST-LINK/V3 embedded debug tool interface.
+This probe allows to flash the board using various tools.
+
+Flashing
+========
+
+The board is configured to be flashed using west `STM32CubeProgrammer`_ runner,
+so its :ref:`installation <stm32cubeprog-flash-host-tools>` is required.
+Version 2.18.0 or later of `STM32CubeProgrammer`_ is required.
+
+Flashing an application to STM32N6570_DK
+------------------------------------------
+
+Connect the STM32N6570_DK to your host computer using the USB port.
+Then build and flash an application. To Flash, BOOT0 pin should be set to 0 and
+BOOT1 to 1. Once flashed, BOOT1 should be set to 0 to run the application.
+Here is an example for the :zephyr:code-sample:`hello_world` application.
+
+Run a serial host program to connect with your Disco board:
+
+.. code-block:: console
+
+   $ minicom -D /dev/ttyACM0
+
+Then build and flash the application.
+
+.. zephyr-app-commands::
+   :zephyr-app: samples/hello_world
+   :board: stm32n6570_dk
+   :goals: build flash
+
+You should see the following message on the console:
+
+.. code-block:: console
+
+   Hello World! stm32n6570_dk/stm32n657xx
+
+
+Debugging
+=========
+
+For now debugging is only available through STM32CubeIDE:
+* Create an empty STM32 project with your board.
+* Create a default debug configuration.
+* In the debug configuration, disable auto build.
+* In the debug configuration, change the C/C++ Application to your zephyr.elf.
```

Comment:
Let's update with the new fancy way as this could avoid issues in some cases.

Reply:
Done